server/src/main/java/org/elasticsearch/common/path/PathTrie.java

```diff
@@ -50,46 +52,39 @@ enum TrieMatchingMode {
         TrieMatchingMode.WILDCARD_ROOT_NODES_ALLOWED
     );
 
-    public interface Decoder {
-        String decode(String value);
-    }
-
-    private final Decoder decoder;
+    private final UnaryOperator<String> decoder;
     private final TrieNode root;
     private T rootValue;
 
     private static final String SEPARATOR = "/";
     private static final String WILDCARD = "*";
 
-    public PathTrie(Decoder decoder) {
+    public PathTrie(UnaryOperator<String> decoder) {
         this.decoder = decoder;
-        root = new TrieNode(SEPARATOR, null, WILDCARD);
+        root = new TrieNode(SEPARATOR, null);
     }
 
-    public class TrieNode {
-        private transient String key;
-        private transient T value;
-        private final String wildcard;
-
-        private transient String namedWildcard;
+    private class TrieNode {
+        private String key;
```

Comment:
we don't need those transient anymore? what was the use of it before?

Reply:
I don't know why these were transient, but the class isn't serializable, so it's not applicable